src/core/dev_api/openvino/core/tensor_util.hpp

```diff
@@ -41,5 +41,20 @@ std::optional<std::vector<T>> to_vector(const ov::Tensor& t) {
         result = ov::op::v0::Constant(t).cast_vector<T>();
     return result;
 }
+
+/// \brief Read a tensor content from a file. Only raw data is loaded.
+/// \param file_name Path to file to read.
+/// \param element_type Element type, when not specified the it is assumed as element::u8.
+/// \param shape Shape for resulting tensor. If provided shape is static, specified number of elements is read only.
+///              File should contain enough bytes, an exception is raised otherwise.
+///              One of the dimensions can be dynamic. In this case it will be determined automatically based on the
+///              length of the file content and `offset`. Default value is [?].
+/// \param offset_in_bytes Read file starting from specified offset. Default is 0. The remining size of the file should
+/// be compatible with shape.
+Tensor read_tensor_from_file(const std::filesystem::path& file_name,
```

Comment:
`read_tensor_from_file` can be interpreted that you are reading an entire tensor representation from a file including data type and shape, at least I would expect that. That's why I used `_data` in my original PR. `_from_file` is redundant as well.